### PR TITLE
[improve][broker]add ServerCnx state check before server handle request

### PR DESCRIPTION
**Pull request by HQebupt**
### Motivation
When the server receives a request, it is supposed to check its own state. like `handleAck\handleCloseProducer\handleConnect\handleFlow\handleProducer`

### Modifications
Add ServerCnx state check when server handle request

### Verifying this change

- [x]  Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.
### Does this pull request potentially affect one of the following parts:

If `yes` was chosen, please highlight the changes

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: (no)
- The schema: (no)
- The default values of configurations: (no)
- The wire protocol: (no)
- The rest endpoints: (no)
- The admin cli options: (no)
- Anything that affects deployment: (no)

### Documentation
Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
- [x] `doc-not-needed`

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
The change looks good me. Could you please help add unit test?

**Comment by HQebupt** [comment]
> The change looks good me. Could you please help add unit test?

Sure.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java [hunk 1]
```diff
@@ -2153,4 +2167,157 @@ public void testHandleAuthResponseWithoutClientVersion() {
         verify(authResponse, times(1)).hasClientVersion();
         verify(authResponse, times(0)).getClientVersion();
     }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    void shouldFailHandleLookup() throws Exception {
+        resetChannel();
```
Comment (AnonHxy) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2173]: I think the test method should be public

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java [hunk 2]
```diff
@@ -2153,4 +2167,157 @@ public void testHandleAuthResponseWithoutClientVersion() {
...
+        Field stateUpdater = ServerCnx.class.getDeclaredField("state");
+        stateUpdater.setAccessible(true);
+        stateUpdater.set(serverCnx, ServerCnx.State.Failed);
```
Comment (AnonHxy) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2176]: This modification will change `serverCnx` state,  which is a common component in the test class. I think it is better to use mock a `ServerCnx` here

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java [hunk 3]
```diff
@@ -2153,4 +2167,157 @@ public void testHandleAuthResponseWithoutClientVersion() {
...
+
+        CommandLookupTopic commandLookupTopic = spy(CommandLookupTopic.class);
+        serverCnx.handleLookup(commandLookupTopic);
```
Comment (AnonHxy) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2179]: Maybe  `serverCnx.handleLookup(any(CommandLookupTopic.class))` is better :)

Comment (HQebupt) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java]: Good

**Review by AnonHxy — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java [hunk 4]
```diff
@@ -2153,4 +2167,139 @@ public void testHandleAuthResponseWithoutClientVersion() {
...
+        ServerCnx serverCnx = mock(ServerCnx.class, CALLS_REAL_METHODS);
+        Field stateUpdater = ServerCnx.class.getDeclaredField("state");
+        stateUpdater.setAccessible(true);
+        stateUpdater.set(serverCnx, ServerCnx.State.Failed);
+        serverCnx.handleTcClientConnectRequest(any(CommandTcClientConnectRequest.class));
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void shouldFailHandleNewTxn() throws Exception {
+        ServerCnx serverCnx = mock(ServerCnx.class, CALLS_REAL_METHODS);
```
Comment (AnonHxy) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2236]: Could we remove  param `CALLS_REAL_METHODS` ?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java [hunk 5]
```diff
@@ -2153,4 +2167,139 @@ public void testHandleAuthResponseWithoutClientVersion() {
...
+        Field stateUpdater = ServerCnx.class.getDeclaredField("state");
+        stateUpdater.setAccessible(true);
+        stateUpdater.set(serverCnx, ServerCnx.State.Failed);
+        serverCnx.handleNewTxn(any(CommandNewTxn.class));
```
Comment (AnonHxy) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2240]: 'any()' is enough, then there is no need to import `CommandNewTxn`

**Comment by AnonHxy** [comment]
LGTM with minor comment

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java (hunk 4 above)
Comment (HQebupt) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2236]: No, we can not due to 'java.lang.IllegalArgumentException' in the constructor of `ServerCnx`

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java (hunk 5 above)
Comment (HQebupt) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java:2240]: OK

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**